### PR TITLE
[ENH] implement logger for first_level_from_bids

### LINKED ISSUE #3686: add a logger to "first_level_from_bids"
I am having to pass this verbose parameter around a bit too much to be able to control whether warnings, messages should be printed...

This should probably be handled better by implementing a logger in another PR.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/3605#discussion_r1152080917_

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3686

Changes proposed in this pull request:

- shows a proof of concept to implement a logger for first_level_from_bids

### CODE REVIEW

File: nilearn/glm/first_level/first_level.py [hunk 1]
```diff
@@ -49,6 +49,11 @@
 from sklearn.cluster import KMeans
 
 
+import logging
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(name="nilearn")
```
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:55]: potentially create a function nilearn/_utils/logger.py ?

File: nilearn/glm/first_level/first_level.py [hunk 2]
```diff
@@ -1527,8 +1511,7 @@ def _make_bids_files_filter(
     if extra_filter and supported_filters:
         for filter_ in extra_filter:
             if filter_[0] not in supported_filters:
-                if verbose:
-                    warn(
+                logger.warning(
```
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:1514]: should warning be handled by the logger as well?

**Comment by Remi-Gau** [comment]
@jeromedockes this is just a quick dirty example of how things could look like with a logger.

Let me know what you think.

**Review by jeromedockes — COMMENTED** [review]
sorry I didn't realise implement a logger meant start using the `logging` standard library module. I think it's a good idea, but then we would probably want to use it consistently throughout the library and plan it a bit more... there is some discussion to be had as some people find `logging` confusing, and we should read through related scikit-learn discussions (linked below).

 and maybe wait for the formatting stuff to be over so we don't have too many widespread changes going on at the same time

for the specific case of 1st level from bids, I wonder if we should have a class to have fewer functions with a lot of parameters, and a way to store some info like the discovered files and produce [reports](https://github.com/nilearn/nilearn/tree/main/nilearn/reporting)?

**Comment by Remi-Gau** [comment]
> sorry I didn't realise implement a logger meant start using the `logging` standard library module. I think it's a good idea, but then we would probably want to use it consistently throughout the library and plan it a bit more... 

yeah the "should we use a more systematic and planned approach to this?" was in the back of my mind and hence why I prefer to draft things here to get early feedback.

Do you think I should maybe rename the issue #3686 to reflect a potential bigger scope for this?

> there is some discussion to be had as some people find `logging` confusing, and we should read through related scikit-learn discussions (linked below).

thanks: will have a look


> and maybe wait for the formatting stuff to be over so we don't have too many widespread changes going on at the same time

Agreed

> for the specific case of 1st level from bids, I wonder if we should have a class to have fewer functions with a lot of parameters, and a way to store some info like the discovered files and produce [reports](https://github.com/nilearn/nilearn/tree/main/nilearn/reporting)?

Will look into this as well.

**Pull request closed without merging**